adam-core/src/test/scala/org/bdgenomics/adam/util/MdTagSuite.scala

```diff
@@ -225,6 +225,131 @@ class MdTagSuite extends FunSuite {
     assert(tag.end === 220)
     assert(tag.toString === "20")
   }
+  
+  test("Get correct matches for mdtag with insertion") {
+    val tag = MdTag("10", 0L, TextCigarCodec.decode("5M3I5M"))
+    assert(tag.end === 9)
+
+    (0 until 9).foreach(locus => assert(tag.isMatch(locus)))
+
+    assert(tag.toString === "10")
+  }
+
+  test("Get correct matches for mdtag with mismatches and insertion") {
+    val tag = MdTag("2A7", 0L, TextCigarCodec.decode("5M3I5M"))
+    assert(tag.end === 9)
+
+    assert(tag.isMatch(0))
+    assert(tag.isMatch(1))
+    assert(tag.mismatches(2) === 'A')
+    (3 until 9).foreach(locus => assert(tag.isMatch(locus)))
+
+    assert(tag.toString === "2A7")
+  }
+
+  test("Get correct matches for mdtag with insertion between mismatches") {
+    val tag = MdTag("2A4A2", 0L, TextCigarCodec.decode("5M3I5M"))
+    assert(tag.end === 9)
+
+    assert(tag.isMatch(0))
+    assert(tag.isMatch(1))
+    assert(tag.mismatches(2L) === 'A')
+
+    assert(tag.mismatches(7L) === 'A')
+    assert(tag.isMatch(8))
+    assert(tag.isMatch(9))
+
+    assert(tag.toString === "2A4A2")
+  }
+
+  test("Get correct matches for mdtag with intron between mismatches") {
+    val tag = MdTag("2A4A2", 0L, TextCigarCodec.decode("5M3N5M"))
+    assert(tag.end === 12)
+
+    assert(tag.isMatch(0))
+    assert(tag.isMatch(1))
+    assert(tag.mismatches(2L) === 'A')
+    assert(tag.isMatch(3))
+    assert(tag.isMatch(4))
+
+    assert(!tag.isMatch(5))
+    assert(!tag.isMatch(6))
+    assert(!tag.isMatch(7))
+
+    assert(tag.isMatch(8))
+    assert(tag.isMatch(9))
+
+    assert(tag.mismatches(10L) === 'A')
+    assert(tag.isMatch(11))
+    assert(tag.isMatch(12))
+
+    assert(tag.toString === "2A4A2")
+  }
+
+  test("Get correct matches for mdtag with intron and deletion between mismatches") {
```

Comment:
Yes! I should was trying to think of more cases, but didn't add the test actually